### PR TITLE
Don't preload completion providers we don't need.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
test vsi please

**Comment by CyrusNajmabadi** [comment]
test this please

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide   Review this plz.  This is an RI blocker.

**Comment by CyrusNajmabadi** [comment]
We used to build a mapping from names to completion providers in the completion service.  Building this mapping required realizing the completion providers.  This was bad as it would cause interactive completion providers to be loaded into the normal (non-interactive) completion service.

We now only add these items to mapping if we actually create teh completion provider.

In order for this to work properly though a lot of tests needed to be fixed up.  That's because the tests would create numerous instances of the completion service, and thus we would have some services would have the mapping filled, while others would not.

Now we create the service at the start of the test and use it throughout the entirety of the test.

File: build/Rulesets/NonShippingProjectRules.ruleset [hunk 1]
```diff
@@ -15,5 +15,6 @@
     <!-- See above comment under CSharp -->
     <Rule Id="RS0003" Action="None" />
     <Rule Id="RS0007" Action="None" />
+    <Rule Id="RS0005" Action="None" />
```
Comment (CyrusNajmabadi) [inline build/Rulesets/NonShippingProjectRules.ruleset]: The "Do not use generic CodeAction.Create to create CodeAction" analyzer is crashing when it analyzed BasicEditorServicesTest.  The analyzer isn't of any value to the test code*, and we need to unblock this RI issue, so i'm just disabling it from running here.

* Then analyzer is about not calling CodeAction.Create directly (and instead creating a subclass of DocumentCodeAction).  That not relevant in test code.

**Comment by CyrusNajmabadi** [comment]
test vsi please

**Comment by CyrusNajmabadi** [comment]
test vsi please

**Comment by CyrusNajmabadi** [comment]
retest vsi0p please

**Comment by CyrusNajmabadi** [comment]
retest vsip0 please

File: build/Rulesets/NonShippingProjectRules.ruleset (hunk 1 above)
Comment (basoundr) [inline build/Rulesets/NonShippingProjectRules.ruleset]: fyi @mavasani

**Comment by basoundr** [comment]
:+1:

**Comment by CyrusNajmabadi** [comment]
Adding @mattwar

**Comment by mattwar** [comment]
:+1:

**Pull request merged**